Added Kibana state

blarghmatey opened the pull request.

blarghmatey committed:
Updated repository.sls to handle multiple products

Updated the repository state to allow for enabling multiple products
from Elastic such as Kibana.

Addresses mitodl/salt-ops#3

blarghmatey committed:
Added installation and management of Kibana

Created a subfolder for containing states pertaining to Kibana because
it is really only relevant in the context of an Elasticsearch
cluster. Installs Kibana using Deb/RPM package and configures it to run
behind an Nginx reverse proxy with SSL.

Addresses mitodl/salt-ops#3

bdero commented on Vagrantfile:
```diff
@@ -6,73 +6,40 @@
+      salt.verbose = true
+      salt.grains_config = "grains/kibana.yml"
+    end
+    kibana.vm.provision "shell", "path": "scripts/testinfra.sh"
```
`"path"` needs to be changed to `path` here (I got a syntax error).

bdero commented:
I'm spinning this up now to give it a test, will try navigating to http://192.168.33.10:80

bdero commented:
Tests pass, TLS redirect with the placement cert works, and kibana responds! Only issue of course is that ES isn't being communicated with, but that's a known WIP issue:
![](http://i.imgur.com/VnuhvDj.png)

blarghmatey committed:
Added handling of auth config for nginx

- Added ability to inject extra config values into nginx
- Updated vagrant setup to properly connect kibana node with elasticsearch node

Addresses mitodl/salt-ops#3

bdero commented:
Rebuilding the boxes to test this out again now.

bdero commented:
I'm having some trouble getting it up and running again - it looks like there were a few states that failed which also cause the tests to fail in turn.

From the output it looks like maybe `makedirs: True` is needed for **configure_kibana_nginx** and **add_file_nginx_auth**?:

```
          ID: configure_kibana_nginx
    Function: file.managed
        Name: /etc/nginx/sites-enabled/kibana
      Result: False
     Comment: Parent directory not present
     Started: 18:23:25.875678
    Duration: 20.65 ms
     Changes:   
----------
          ID: remove_default_nginx_config
    Function: file.absent
        Name: /etc/nginx/sites-enabled/default
      Result: True
     Comment: File /etc/nginx/sites-enabled/default is not present
     Started: 18:23:25.896817
    Duration: 1.428 ms
     Changes:   
----------
          ID: kibana_service
    Function: service.running
        Name: kibana
      Result: False
     Comment: One or more requisite failed: elasticsearch.kibana.configure_kibana
     Started: 
    Duration: 
     Changes:   
----------
          ID: kibana_nginx_service
    Function: service.running
        Name: nginx
      Result: False
     Comment: One or more requisite failed: elasticsearch.kibana.configure_kibana_nginx
     Started: 
    Duration: 
     Changes:   
----------
          ID: add_file_nginx_auth
    Function: file.managed
        Name: /etc/nginx/kibana_users
      Result: False
     Comment: Parent directory not present
     Started: 18:23:25.899670
    Duration: 1.439 ms
     Changes:   
----------
          ID: add_extra_config_to_nginx
    Function: file.blockreplace
        Name: /etc/n
ginx/sites-enabled/kibana
      Result: False
     Comment: One or more requisite failed: elasticsearch.kibana.nginx_extra_config.add_file_nginx_auth
     Started: 
    Duration: 
     Changes:   

Summary for local
-------------
Succeeded: 21 (changed=19)
Failed:     9
-------------
Total states run:     30
```

blarghmatey committed:
Updated rendering of YAML configs

Switched to rendering the pillar data directly as YAML in the contents
of the target files, bypassing the templates. This is to address issues
with line endings not being represented properly.

Addresses mitodl/salt-ops#3

blarghmatey committed:
Added missing java dependency

Added java dependency to kibana map.jinja for running elasticsearch

blarghmatey committed:
Updated nginx extra config

Updated yaml for add_file to properly handle multi-line contents

blarghmatey committed:
Pkg state updates

- Changed elasticsearch and kibana to be `pkg.installed` instead of
  `pkg.latest`
- Explicitly included elasticsearch state for kibana es_client_nodes

blarghmatey committed:
Restart elasticsearch after plugin install

- Added watch statement to restart elasticsearch when installing plugins
